packages/gp2-components/src/organisms/__tests__/ResourceModal.test.tsx

```diff
@@ -119,14 +165,109 @@ describe('ResourceModal', () => {
     await waitFor(() => expect(saveButton).toBeEnabled());
   });
   it('selecting a type is required', () => {
+    renderResourseModal();
     save();
     expect(screen.getByText(/please enter a valid type/i)).toBeVisible();
   });
 
   it('a title and link is required', () => {
+    renderResourseModal();
     enterType('Link');
     save();
     expect(screen.getByText(/please enter a title/i)).toBeVisible();
     expect(screen.getByText(/please enter a valid link/i)).toBeVisible();
   });
+
+  it('the dialog shows when the user changes the resource information and cancels the action', () => {
+    getUserConfirmation = jest.fn((_message, cb) => cb(true));
+    history = createMemoryHistory({ getUserConfirmation });
+
+    const props = {
+      modalTitle: editModalInfo.title,
+      modalDescription: editModalInfo.description,
+      type: 'Note' as const,
+      title: 'A title',
+      description: 'A description',
+    };
+
+    render(
+      <Router history={history}>
+        <ResourceModal {...defaultProps} {...props} />
+      </Router>,
+    );
```

Comment:
We keep repeating this code. Let't make a helper function with defaults which can be overridden when we need.